src/Commands/Extension/AzCommand.cs

```diff
@@ -72,43 +64,44 @@ protected override AzArguments BindArguments(ParseResult parseResult)
 
     private static string? FindAzCliPath()
     {
+        string executableName = "az";
+
         // Return cached path if available and still exists
         if (!string.IsNullOrEmpty(_cachedAzPath) && File.Exists(_cachedAzPath))
         {
             return _cachedAzPath;
         }
 
-        var isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
-        var searchPaths = new List<string>(AzureCliPaths);
-
-        // Add PATH environment directories
-        if (Environment.GetEnvironmentVariable("PATH")?.Split(Path.PathSeparator) is { } pathDirs)
-        {
-            searchPaths.AddRange(pathDirs);
-        }
+        var pathEnv = Environment.GetEnvironmentVariable("PATH");
+        if (string.IsNullOrEmpty(pathEnv))
+            return null;
 
-        foreach (var dir in searchPaths.Where(d => !string.IsNullOrEmpty(d)))
+        string[] paths = pathEnv.Split(Path.PathSeparator);
+        foreach (string path in paths)
         {
-            if (isWindows)
+            string fullPath = Path.Combine(path.Trim(), executableName);
+            if (File.Exists(fullPath))
             {
-                var cmdPath = Path.Combine(dir, "az.cmd");
-                if (File.Exists(cmdPath))
+                var isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+                if (isWindows)
                 {
-                    _cachedAzPath = cmdPath;
-                    return cmdPath;
-                }
-            }
-            else
-            {
-                var fullPath = Path.Combine(dir, "az");
-                if (File.Exists(fullPath))
-                {
-                    _cachedAzPath = fullPath;
-                    return fullPath;
+                    string exePath = fullPath + ".cmd";
```

Comment:
use System.IO.Path.ChangeExtension() here instead of string concatenation please.

https://learn.microsoft.com/en-us/dotnet/api/system.io.path.changeextension?view=net-9.0